File: euler_049.py
from time import time
from primePy import primes

def are_perms(p,q):
    return sorted(str(p)) == sorted(str(q))

start = time()

p_list = list(primes.between(1000,10000))


for p in p_list:
    for q in p_list:
        if p < q:
            if q + (q - p) in p_list:
                if are_perms(p,q) and are_perms(p, 2 * q - p):
                    print((p, q, 2 * q - p), str(p) + str(q) + str(2 * q - p))





stop = time()
print("Time: " + str(stop-start))

File: euler_049.js
const start = Date.now();

// Primes
const upperBound = 10000;

let sieve = Array(upperBound);
sieve.fill(true);
sieve[0] = false;
sieve[1] = false;

let primes = [];

for (let c = 2;c < upperBound; c++) {
    if (sieve[c]) {
        if (c > 1000) {primes.push(c);}
        let j = 2;
        while (c * j < upperBound) {
            sieve[c * j] = false;
            j++;
        }
    }
}
// End of primes

const arePermutations = (a,b,c) => {
    return (a.toString().split("").sort().join("") == b.toString().split("").sort().join("") && a.toString().split("").sort().join("") == c.toString().split("").sort().join(""));
}

let arithmetic = [];
let stillLooking = true;


for (const p of primes) {
  for (const q of primes.filter((x) => x > p)) {
    let diff = q - p;
    if (p !== 1487 && sieve[q + diff] && arePermutations(p, q, q + diff)) {
      arithmetic.push(p);
      arithmetic.push(q);
      arithmetic.push(q + diff);
    }
  }
}  


const res = arithmetic.reduce((p,c) => p + c.toString(),"");


const end = Date.now();

console.log("Execute time: ", end - start, " The first of the numbers is: ", res);